tests/functional/partial_parsing/test_pp_vars.py

```diff
@@ -334,7 +334,7 @@ def test_profile_env_vars(self, project):
         with pytest.raises(FailedToConnectError):
             run_dbt(["run"], expect_pass=False)
 
-        log_output = Path(project.project_root, "logs", "dbt.log").read_text()
+        log_output = Path(logs_dir, "dbt.log").read_text()
```

Comment:
This test in main has the code:
```
        # Change env_vars, the user doesn't exist, this should fail
        os.environ["ENV_VAR_USER"] = "fake_user"                     
        (results, log_output) = run_dbt_and_capture(["run"], expect_pass=False)
        assert "env vars used in profiles.yml have changed" in log_output
        manifest = get_manifest(project.project_root)                                       
        assert env_vars_checksum != manifest.state_check.profile_env_vars_hash.checksum            
```

Is there some reason that this test is now throwing an error in click-cli when it doesn't in main? And is there some reason that it's not doing "run_dbt_and_capture"?  In a general way it's better to read the output for a single command using run_dbt_and_capture rather than reading the entire log file, because there might be other commands running which will confuse the checks for output in the log for this particular command. For this particular test, it doesn't look like it would matter, but in a general way those two ways of getting the logs are not exactly equivalent.

Reply:
The test was originally changed in [this PR](https://github.com/dbt-labs/dbt-core/pull/6856/files) to read the logs this way instead of via `run_dbt_and_capture`. 

> Is there some reason that this test is now throwing an error in click-cli when it doesn't in main?

It does appear that this exception is swallowed on main but not on the feature branch. It's not  clear why that's the case or whether it's something to do with the test setup/framework or the code, but will keep looking into it.

Reply:
I think I remember being surprised that it didn't raise an error. Would be kind of nice to know why, but It's probably not worth spending time on, since it does seem like the branch is right and main is wrong.